### PR TITLE
[server] Make JWT Session rolling through refresh endpoint WEB-418

### PR DESCRIPTION
**Pull request by easyCZ**
## Description

If the token is older than 1h, the refresh endpoint will also update it.

## Related Issue(s)

* Depends on https://github.com/gitpod-io/gitpod/pull/17797

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - mp-server-c9ccfb03dd</li>
	<li><b>🔗 URL</b> - <a href="https://mp-server-c9ccfb03dd.preview.gitpod-dev.com/workspaces" target="_blank">mp-server-c9ccfb03dd.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
</ul>

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [x] with-ws-manager-mk2
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`
</details>
 
/hold

### CODE REVIEW

File: components/server/src/session-handler.ts [hunk 1]
```diff
@@ -87,15 +87,40 @@ export class SessionHandler {
...
+                            reportJWTCookieIssued();
+                            res.status(200);
+                            res.send("Refreshed JWT cookie issued.");
+                            return;
+                        }
+
+                        res.status(200);
+                        res.send("User session already has a valid JWT session.");
+                    } catch (err) {
+                        reportJWTCookieIssued();
```
Comment (AlexTugarev) [inline components/server/src/session-handler.ts:119]: this might be wrong here, no?

Comment (easyCZ) [reply components/server/src/session-handler.ts:119]: Ah yes, good catch.

**Review by AlexTugarev — APPROVED** [review]
Code changes look good. 
Only question about metric instruction [here](https://github.com/gitpod-io/gitpod/pull/17798/files#r1217622684).

Didn't test though 🙈

**Pull request merged**